packages/router/router.ts

```diff
@@ -463,26 +463,8 @@ export function createRouter(init: RouterInit): Router {
     initialErrors = { [route.id]: error };
   }
 
-  // If we received hydration data without errors - detect if any matched
-  // routes with loaders did not get provided loaderData, and if so launch an
-  // initial data re-load to fetch everything
-  let foundMissingHydrationData =
-    init.hydrationData?.errors == null &&
-    init.hydrationData?.loaderData != null &&
-    initialMatches
-      .filter((m) => m.route.loader)
-      .some((m) => init.hydrationData?.loaderData?.[m.route.id] === undefined);
-
-  if (foundMissingHydrationData) {
-    console.warn(
-      `The provided hydration data did not find loaderData for all matched ` +
-        `routes with loaders.  Performing a full initial data load`
```

Comment:
Can't do this in a remix context unfortunately.  Every route has a `loader` for JS preloading - but not every route's loader will fetch data or put an entry in `loaderData`.  I guess we could require remix to return `null` as it's `loaderData` in that case?

```
hydrationData = {
    loaderData: {
        routeWithoutLoader: null
    }
}
```

What do you think?